src/code.c

```diff
@@ -874,11 +854,21 @@ void CodeFuncExprEnd(UInt nr)
     else {
         stat1 = PopStat();
         PushStat( stat1 );
-        if ( TNUM_STAT(stat1) != T_RETURN_VOID
-          && TNUM_STAT(stat1) != T_RETURN_OBJ )
-        {
-            CodeReturnVoidWhichIsNotProfiled();
-            nr++;
+        while( 1 ){
```

Comment:
Code formatting!
```suggestion
        while (1) {
```

Reply:
Also, this code *really* needs to be commented. Both what it is doing (see my suggestion below), but also more importantly, *why* it is doing it. Which I don't know right now...

Reply:
Why doesn't it suffice to insert the following into the existing code, to achieve the same effect?
```
  // find the last statement
  while (T_SEQ_STAT <= TNUM_STAT(stat1) && TNUM_STAT(stat1) <= T_SEQ_STAT7) {
    TNUM_STAT(stat1) <= T_SEQ_STAT7) {
    UInt size = SIZE_STAT(stat1) / sizeof(Stat);
    stat1 = READ_STAT(stat1, size - 1);
  }
```

Reply:
Well, logically, this does exactly the same, except it is not intertwined with the addition of the `return void` to the statement.

Reply:
About the why: The thing is that if we reread a syntax tree, the statements are already nested, i.e., if there are more than 7 statements in the body, the last one is already a new `SEQ_STAT*`. This ensures that the `return;` is only added if there is really no `return` in the last statement.

When a function is coded, the statements are packed into nested statements later.

I discovered this problem when I realized that "long" functions get an additional `return;`